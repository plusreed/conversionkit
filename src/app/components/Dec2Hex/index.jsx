import React, { Component } from 'react';
import { hex } from '../../../lib/ConversionCore';

export default class Dec2Hex extends Component {
    constructor() {
        super();

        this.state = {
            number: "",
            hexRepresentation: "",
        }

        this.convertToHex = this.convertToHex.bind(this);
    }

    convertToHex(event) {
        this.setState({
            number: parseInt(event.target.value),
            hexRepresentation: hex(parseInt(event.target.value)),
        });
    }

    render() {
        return (
            <div className="p-4">
                <p className="text-sm mb-2 text-gray-600">Hex representation of {this.state.number === "" || isNaN(this.state.number) ? 'Nothing' : this.state.number} is: {this.state.hexRepresentation === "" || this.state.hexRepresentation <= -1 ? 'Nothing' : this.state.hexRepresentation}</p>
                <form>
                    <input
                        className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
                        type="number"
                        placeholder="Decimal here..."
                        value={this.state.number || ""}
                        onChange={this.convertToHex}
                    />
                </form>
            </div>
        )
    }
};
