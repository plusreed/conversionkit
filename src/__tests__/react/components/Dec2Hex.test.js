import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Dec2Hex from '../../../app/components/Dec2Hex';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Dec2Hex />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const tree = renderer
    .create(<Dec2Hex />)
    .toJSON()

  expect(tree).toMatchSnapshot()
});