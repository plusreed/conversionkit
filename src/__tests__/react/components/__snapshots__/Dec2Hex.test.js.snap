// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="p-4"
>
  <p
    className="text-sm mb-2 text-gray-600"
  >
    Hex representation of 
    Nothing
     is: 
    Nothing
  </p>
  <form>
    <input
      className="bg-white focus:outline-none focus:shadow-outline border border-gray-300 rounded-lg py-2 px-4 block w-full appearance-none leading-normal"
      onChange={[Function]}
      placeholder="Decimal here..."
      type="number"
      value=""
    />
  </form>
</div>
`;
